frontend/js/src/utils/QueryClient.ts

```diff
@@ -0,0 +1,13 @@
+import { QueryClient } from "@tanstack/react-query";
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+      retry: 1,
+      staleTime: 1000 * 60 * 5,
```

Comment:
Sounds reasonable for here.

For pages like the user dashboard, as soon as we have a timestamp (i.e not on the first page of my listens) we can cache the results for a loooong time since the API results should be the same (barring deleted listens, the deletion process is run once an hour I think?)